test/lib/pgBackRestTest/Module/Manifest/ManifestAllTest.pm

```diff
@@ -0,0 +1,1088 @@
+
+        # SubKey required but has not been set
+        #---------------------------------------------------------------------------------------------------------------------------
+        $self->testException(sub {$oManifest->boolGet(MANIFEST_SECTION_TARGET_FILE, BOGUS, MANIFEST_SUBKEY_CHECKSUM_PAGE)},
+            ERROR_ASSERT, "strSection '" . MANIFEST_SECTION_TARGET_FILE . "', strKey '" . BOGUS . "', strSubKey '" .
+            MANIFEST_SUBKEY_CHECKSUM_PAGE . "' is required but not defined");
+
+        # SubKey not required and not set
+        #---------------------------------------------------------------------------------------------------------------------------
+# CSHANG Why would we allow this?
+        $self->testResult(sub {$oManifest->boolGet(MANIFEST_SECTION_TARGET_FILE, BOGUS, MANIFEST_SUBKEY_CHECKSUM_PAGE, false)},
+            false, 'boolGet() - false');
+
+        # Set and get a boolean value
+        #---------------------------------------------------------------------------------------------------------------------------
+        $oManifest->boolSet(MANIFEST_SECTION_TARGET_FILE, BOGUS, MANIFEST_SUBKEY_CHECKSUM_PAGE, true);
+        $self->testResult(sub {$oManifest->boolGet(MANIFEST_SECTION_TARGET_FILE, BOGUS, MANIFEST_SUBKEY_CHECKSUM_PAGE)}, true,
+            'boolGet() - true');
+
+        # Get default boolean value
+        #---------------------------------------------------------------------------------------------------------------------------
+# CSHANG But this doesn't actually set the default value, so should we be allowing this? Why?
+        $self->testResult(sub {$oManifest->boolGet(MANIFEST_SECTION_BACKUP_OPTION, MANIFEST_KEY_COMPRESS, undef, false, true)},
+            true, 'boolGet() - default true');
+
+        # Get but don't return a default boolean value even if there is supposed to be one
+        #---------------------------------------------------------------------------------------------------------------------------
+# CSHANG Is this right? Should the calling routine be passing whether the option has a default or not?
+        $self->testResult(sub {$oManifest->boolGet(MANIFEST_SECTION_BACKUP_OPTION, MANIFEST_KEY_COMPRESS, undef, false, false)},
+            false, 'boolGet() - default false');
+
+        # Get default numeric value
+        #---------------------------------------------------------------------------------------------------------------------------
+# CSHANG But this doesn't actually set the default value, so should we be allowing this? Why?
+        $self->testResult(sub {$oManifest->numericGet(MANIFEST_SECTION_TARGET_FILE, BOGUS, MANIFEST_SUBKEY_SIZE, false, 0)},
+            0, 'numericGet() - default 0');
+
+        # Coverage for $self->SUPER::get("${strSection}:default"
+        #---------------------------------------------------------------------------------------------------------------------------
+# CSHANG This returns undef because required = false - is that good?
```

Comment:
Yes, this is OK - since ->get() is an override for the underlying Ini function is needs to expose the functionality of that function, except for special cases.